server/core/src/access_time_queue.cpp

```diff
@@ -0,0 +1,137 @@
+#include "irods/access_time_queue.hpp"
+
+#include "irods/irods_exception.hpp"
+#include "irods/rodsErrorTable.h"
+
+#include <boost/interprocess/ipc/message_queue.hpp>
+#include <fmt/format.h>
+#include <unistd.h>
+
+#include <cctype>
+#include <chrono>
+#include <memory>
+#include <string>
+
+namespace
+{
+    //
+    // Global Variables
+    //
+
+    // On initialization, holds the PID of the process that initialized the access time queue.
+    // This ensures that only the process that initialized the system can deinitialize it.
+    pid_t g_owner_pid; // NOLINT(cppcoreguidelines-avoid-non-const-global-variables)
+
+    // Holds the name of the message queue which will exist in shared memory.
+    std::string g_mq_name; // NOLINT(cppcoreguidelines-avoid-non-const-global-variables)
+
+    // The pointer to the message queue which will exist in shared memory.
+    // Allocating on the heap allows us to know when the message queue is constructed/destructed.
+    // NOLINTNEXTLINE(cppcoreguidelines-avoid-non-const-global-variables)
+    std::unique_ptr<boost::interprocess::message_queue> g_mq;
+
+    // Returns true if the string starts with an alphabet or underscore followed by alphanumeric
+    // characters. The string is expected to contain only ASCII character sequences. Anything else
```

Comment:
I saw `[_a-zA-Z][_a-zA-Z0-9]*` somewhere... My understanding of the requirements is more like this:
```suggestion
    // Returns true if the string starts with an alphabetic character or underscore, optionally followed by one or more alphanumeric
    // characters and/or underscores. The string is expected to contain only ASCII character sequences. Anything else
```
Is that right? Maybe I'm still sleepy.

Reply:
That's correct.

Will fix.

Reply:
oh, 'starting with'... i see... carry on.

Reply:
Done.